DDF-3650 Create Search Form View and refactor existing functionality

bdeining opened the pull request:
#### What does this PR do?
Create Template View and refactor existing functionality.  Currently fixing some CSS issues with the dropdown view.
#### Who is reviewing it? 
@mackncheesiest @Lambeaux 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@andrewkfiedler
@jlcsmith
@millerw8
#### How should this be tested? (List steps with links to updated documentation)
Build / Install / Create Query / Mess around with templates
#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-3650](https://codice.atlassian.net/browse/DDF-3650)
#### Screenshots (if appropriate)
![screen shot 2018-02-26 at 11 54 50 am](https://user-images.githubusercontent.com/6551038/36691219-c96e3538-1af1-11e8-8eba-30ce68a7f314.png)
![screen shot 2018-02-26 at 11 54 57 am](https://user-images.githubusercontent.com/6551038/36691220-c9859ebc-1af1-11e8-8757-80468908af5a.png)
![screen shot 2018-02-26 at 11 55 28 am](https://user-images.githubusercontent.com/6551038/36691221-c99d7bae-1af1-11e8-8a83-df4b0abd9dea.png)
![screen shot 2018-02-26 at 11 55 33 am](https://user-images.githubusercontent.com/6551038/36691223-c9cacd0c-1af1-11e8-8460-d21890ad8e54.png)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

djblue commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-templates/template-collection.js:
```diff
@@ -0,0 +1,29 @@
+ *
+ **/
+/*global define */
+define([
```
I think you should use `require` here instead of `define`.

bdeining replied:
done

michaelmenousek commented:
-1 at the moment.

Please say "Use another Search Form" and refer to these as "Search Forms" rather than "Templates"  -- also please make the cards and window larger with more "white space".  They don't look selectable.

The grey color used in "Basic" and "Text" is too light and looks greyed out.

Please say "Search Form Browser" in the title bar (to align with future changes)

bdeining commented:
@michaelmenousek addressed your comments.  Here is an updated screenshot that also shows highlighting.  The text is greyed out because nothing was highlighted in the previous screenshot.
![screen shot 2018-02-27 at 10 22 57 am](https://user-images.githubusercontent.com/6551038/36743898-a91930fa-1ba8-11e8-860d-bc4197ea3c3b.png)

Schachte commented on catalog/ui/catalog-ui-search/src/main/webapp/component/dropdown/search-type/dropdown.search-type.hbs:
```diff
@@ -14,5 +14,5 @@
 </div>
 <div class="interaction-text">
-    Change Type
+    Use another Search Form
```
nit: Should A in "another" be capitalized since everything is title case?

michaelmenousek replied:
thanks @bdeining!  I am +1 now.

bdeining replied:
fixed capitalization

Schachte commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-interactions/search-interactions.hbs:
```diff
@@ -11,29 +11,20 @@
+<div class="interaction interaction-type" title="Change the form used to construct the search." data-help="Change the form used to construct the search."></div>
+
+<div class="interaction interaction-type-advanced interaction-form" title="Change the form used to construct the search." data-help="Change the form used to construct the search.">
+    <div class="interaction-icon fa fa-question-circle"></div>
```
nit: question mark seems a bit odd for "advanced" maybe something like `<i class="fas fa-search"></i>`

bdeining replied:
yeah, seems more consistent with fa-search

Schachte commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-forms/search-forms.view.js:
```diff
@@ -0,0 +1,39 @@
+    onRender: function () {
+        this.tabsContent.show(new SearchFormsView({
+            model : this.model
+        }));
```
Should we have a triggerDropdown() here? Otherwise, if we select a template the dropdown remains open and feels awkward.

bdeining replied:
I agree; looking into this now

bdeining replied:
Fixed

bdeining commented:
@Variadicism fixed inconsistent spacing

Variadicism commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-forms/search-form-view.js:
```diff
@@ -0,0 +1,51 @@
+                this.model.queryModel.set('type', 'text');
+                user.getQuerySettings().set('type', 'text');
+                break;
+        }
```
This `switch` statement can be simplified to the following `if` statement:
```
const type = this.model.get('type');
if (type === 'basic' || type === 'text') {
    this.model.queryModel.set('type', type);
    user.getQuerySettings().set('type', type);
}
```

bdeining replied:
we might want to leave this as a switch as more cases will likely be added in the future

Variadicism commented:
@bdeining Thanks. I wasn't sure if spacing was considered significant enough for code review comments. If so, I've marked a couple other spots with inconsistent spacing, but they're :pencil2:.

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-forms/search-forms.less:
```diff
@@ -0,0 +1,32 @@
+    margin: @minimumSpacing;
+    width: 200px;
+    height: 150px;
+    text-align:center;
```
Add `vertical-align: top` to keep the choices aligned correctly.

bdeining replied:
done

andrewkfiedler commented:
We should update the folder name for `search-forms` to be `search-form`, as well as update the name for `search-form-collection.js` to be `search-form.collection.js`.

andrewkfiedler commented:
Move the current `search-form-view.js` into another folder and rename it to differentiate it as the search form selector.  Maybe `search-form-selector.view.js` in `search-form-selector` folder.

bdeining commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-forms/search-form-view.js:
```diff
@@ -0,0 +1,51 @@
+        }
+        user.savePreferences();
+        this.triggerCloseDropdown();
+        this.model.queryModel.trigger('closeDropDown');
```
moved inside of `triggerCloseDropdown`

bdeining commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-forms/search-form.js:
```diff
@@ -0,0 +1,29 @@
+    defaults: {
+        name  :"A Search Form",
+        createdBy  :"admin",
+        createdOn  :"30 October 2018",
```
fixed

bdeining commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-forms/search-forms.less:
```diff
@@ -0,0 +1,32 @@
+@{customElementNamespace}search-form-system {
+
+    .search-form-system-title {
+        text-align:center;
```
fixed

bdeining commented:
@andrewkfiedler addressed your comments

andrewkfiedler commented:
Thanks, I'll take another look, I got pulled away to work on something while I was reviewing.

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.collection.view.js:
```diff
@@ -0,0 +1,30 @@
+     childView: SearchFormView,
+     tagName: CustomElements.register('my-search-forms'),
+     className: 'is-list is-inline has-list-highlighting',
+     initialize: function(options) {
```
An easier way to pass these options on to the childView will be to use `childViewOptions`.  We do something similar in https://github.com/codice/ddf/blob/430880eb5d2c334ed74c78adb04d16e472225b82/catalog/ui/catalog-ui-search/src/main/webapp/component/result-item/result-item.collection.view.js#L37

bdeining replied:
Done

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form-collection.js:
```diff
@@ -0,0 +1,41 @@
+/**
```
We should update the filename to `search-form.collection.js`.

bdeining replied:
done

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form-collection.js:
```diff
@@ -0,0 +1,41 @@
+       this.add(new SearchForm({type: "basic", queryModel: options[0].queryModel}));
+       this.add(new SearchForm({type: "text", queryModel: options[0].queryModel}));
+   },
+   getCustomQueryForms: function() {
```
Is this for a later PR to utilize?

bdeining replied:
yes, there will be a follow-on PR that will utilize this

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.js:
```diff
@@ -0,0 +1,29 @@
+
+ module.exports = Backbone.Model.extend({
+    defaults: {
+        name: "A Search Form",
```
We're going to need an id on these as well.

bdeining replied:
The ID will be passed from the backend; these defaults will probably be removed in a future iteration

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.js:
```diff
@@ -0,0 +1,29 @@
+        type: "custom"
+    },
+    initialize: function(options) {
+        if (options.type) {
```
Backbone already allows settings model values upon creation by passing them in.

bdeining replied:
fixed

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.js:
```diff
@@ -0,0 +1,24 @@
+    defaults: {
+        name: "A Search Form",
+        createdBy: "admin",
+        createdOn: "30 October 2018",
```
We'll need to account for these coming from the backend in a more standard format (unless normalization is happening in between).  Right now the views are using these values directly, when in reality they'll likely need to do some sort of transformation.

bdeining replied:
From the backend this will come in as Unix Epoch time; added logic to transform into a human readable form

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.js:
```diff
@@ -0,0 +1,24 @@
+        createdBy: "admin",
+        createdOn: "30 October 2018",
+        type: "custom",
+        formTemplate: "{}"
```
Is the actual template going to be returned, or is it going to be an id that lets us know how to look up the template when the user wants to use it?

bdeining replied:
The current implementation is using a `formTemplate` that has been returned from the backend.

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/filter/filter.view.js:
```diff
@@ -67,10 +67,12 @@ define([
                 break;
             default:
-                if (value[0].constructor === Object) {
+               if (value[0] == null) {
```
Why do we need a null check here?

bdeining replied:
The response coming back from the `Search Form` backend won't populate values in some cases, since it is a template.  This is currently being reworked to have a filter specific to `Search Forms`, so this will be OBE in the next commit

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/query-add/query-add.view.js:
```diff
@@ -51,6 +52,7 @@ module.exports = Marionette.LayoutView.extend({
         this.$el.toggleClass('is-text', false);
         this.$el.toggleClass('is-basic', false);
         this.$el.toggleClass('is-advanced', false);
+        this.$el.toggleClass('is-custom', false);
```
I don't think we need to toggle these classes (`is-text`, etc.), nothing appears to utilize them.

bdeining replied:
Removed

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/query-custom/query-custom.view.js:
```diff
@@ -0,0 +1,104 @@
+/**
```
We should be able to reuse `query-advanced` component here.

bdeining replied:
Done

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/query-editor/query-editor.view.js:
```diff
@@ -130,6 +131,7 @@ module.exports = Marionette.LayoutView.extend({
         this.$el.toggleClass('is-text', false);
         this.$el.toggleClass('is-basic', false);
         this.$el.toggleClass('is-advanced', false);
+        this.$el.toggleClass('is-custom', false);
```
Similar to above comment, doesn't look like we use these classes.  It might be that we could refactor `query-editor` and `query-add` to use the same component.

bdeining replied:
Removed unused classes; attempted to refactor to use the same component, but it wasn't trivial.  Since this is outside the scope of templates, I recommend we make a tech debt ticket for this change as it exists in master already.

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form-selector/search-forms-selector.view.js:
```diff
@@ -0,0 +1,39 @@
+/**
```
We should update the folder name and the component name to match (one is plural for forms and the other isn't).

In addition, we should update the tag name used to be the same as the folder name (`search-form-selector`).

bdeining replied:
renamed

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form-selector/search-forms-selector.view.js:
```diff
@@ -0,0 +1,39 @@
+ *
+ **/
+/*global require*/
+var _ = require('underscore');
```
There are some unused imports here (_, $)

bdeining replied:
removed

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form-selector/search-forms-selector.hbs:
```diff
@@ -0,0 +1,16 @@
+ **/
+ --}}
+
+<div class="search-form-system-title"><span>Search Form Browser</span></div>
```
Since we namespace components and their css (less), we can use shorter classnames.

bdeining replied:
refactored class names

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form-selector/search-forms-selector.hbs:
```diff
@@ -0,0 +1,16 @@
+ **/
+ --}}
+
+<div class="search-form-system-title"><span>Search Form Browser</span></div>
```
No need for the enclosing span here.

bdeining replied:
removed

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form-selector/search-forms-selector.view.js:
```diff
@@ -0,0 +1,39 @@
+        tabsContent: {
+            selector: '.search-form-system-content'
+        },
+        tabsTitle: {
```
No need for the title to be a region since it's not getting replaced with another view.

bdeining replied:
removed extra region

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form-loading.view.js:
```diff
@@ -0,0 +1,28 @@
+/**
```
Unless we're going to do something fancy with the loading, we can probably put the spinner directly in the view that needs it (and hide and show it there).

bdeining replied:
done

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.collection.js:
```diff
@@ -0,0 +1,50 @@
+/**
```
We should refactor this to make it so the non-custom (predefined) templates can be changed to immediately even though the custom ones are still loading.  

We should consider if we want to keep the results of the get request around so that we don't have to refetch them every time the search form selector is closed and reopened.

bdeining replied:
refactored to return non-custom templates immediately

Schachte replied:
Is it worth considering caching custom templates in local storage to improve load time? Might improve user experience.

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.collection.js:
```diff
@@ -0,0 +1,50 @@
+   },
+   saveExport: function(data) {
+        var that = this;
+        $.each(data, function(index, value) {
```
Backbone allows you to send an array of models into the add function, so we can remove this $.each.

bdeining replied:
Don't we still need to create the models to be added to the collection in an `$.each`?

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.collection.js:
```diff
@@ -0,0 +1,50 @@
+            url: '/search/catalog/internal/forms/query',
+            contentType: 'application/json',
+            success: function (data) {
+                that.saveExport(data)
```
There are a few other ways to do this without an extra variable. 

Ajax has an option for context we could use.  Alternatively we can use .bind.  Or we can use arrow functions to bind the context.

bdeining replied:
Added `context: this`

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.collection.js:
```diff
@@ -0,0 +1,50 @@
+                that.saveExport(data)
+            },
+            failure: function () {
+                this.$el.trigger("doneLoading");
```
This will trigger a dom event instead of an event on the collection.

bdeining replied:
removed $el

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.collection.js:
```diff
@@ -0,0 +1,50 @@
+        $.each(data, function(index, value) {
+            that.add(new SearchForm({id: value.id, name: value.title, type: 'custom', formTemplate: value}));
+        });
+        this.trigger("doneLoading");
```
It would be better to have this be a property on a model.  Maybe do an associated model for search form collections, where the collection is nested inside.

bdeining replied:
Curious :  what makes an associated model better in this case?

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.temp.view.js:
```diff
@@ -0,0 +1,37 @@
+/**
```
Can we rename this to `search-form.collection.view.js`?

bdeining replied:
That name exists already; renamed to be `search-form-tab-container.view.js`

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.temp.view.js:
```diff
@@ -0,0 +1,37 @@
+     tagName: CustomElements.register('my-search-forms'),
+     className: 'is-list is-inline has-list-highlighting',
+     initialize: function(options) {
+        this.collection = new SearchFormCollection(null, [{queryModel: options.model}]);
```
With the childViewOptions being used we don't need to pass in anything here.

bdeining replied:
Done

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.view.js:
```diff
@@ -0,0 +1,64 @@
+        'click': 'changeView'
+    },
+    initialize: function(options) {
+        this.queryModel = options.queryModel;
```
We don't need to store this off in a variable because it can be accessed in other areas of the view through `this.options.queryModel`.

bdeining replied:
Done

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.view.js:
```diff
@@ -0,0 +1,64 @@
+        }
+    },
+    changeView: function() {
+        this.queryModel.set('formTemplate', this.model.get('formTemplate'));
```
We should be saving the id of the template instead, formTemplate isn't guaranteed to be static over time.

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.view.js:
```diff
@@ -0,0 +1,64 @@
+                user.getQuerySettings().set('type', 'text');
+                break;
+            case 'custom':
+                this.queryModel.trigger('change:type');
```
Setting the type should trigger this automatically.

andrewkfiedler replied:
I'm guessing this is because it won't change when you go between two different custom templates.  Switching to utilize id as I mentioned in other comments will alleviate this issue.

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-forms.less:
```diff
@@ -0,0 +1,36 @@
+@{customElementNamespace}search-form {
```
We should rename this file to be `search-form.less`.

bdeining replied:
Done

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-forms.less:
```diff
@@ -0,0 +1,36 @@
+    display: inline-block;
+    padding: @minimumSpacing;
+    margin: @minimumSpacing;
+    width: 300px;
```
Can we update these to use a multiple of minimumButtonSize?  Then it will respond to the font size settings and the spacing settings.

bdeining replied:
Done

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-forms.less:
```diff
@@ -0,0 +1,36 @@
+    width: 300px;
+    height: 125px;
+    text-align: center;
+    vertical-align: middle;
```
The vertical alignment on the search-form types should match (either both middle or both top, otherwise they might appear offset from one another).

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-forms.less:
```diff
@@ -0,0 +1,36 @@
+    margin: @minimumSpacing;
+    width: 300px;
+    height: 125px;
+    text-align: center;
```
I think we can get rid of the css that isn't text-align center, since that is already covered by the less above for `search-form`s.

bdeining replied:
Done

mackncheesiest commented on catalog/ui/catalog-ui-search/src/main/webapp/component/search-interactions/search-interactions.less:
```diff
@@ -51,22 +45,4 @@
-  }
-}
-
-@{customElementNamespace}search-interactions.is-basic {
```
Why did we get rid of the quick "basic" and "advanced" selectors in the search-interactions menu?

bdeining replied:
This was based on design discussions; these were moved under templates

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/tabs/search-form/tabs-search-form.js:
```diff
@@ -0,0 +1,24 @@
+ /*global require*/
+ var _ = require('underscore');
+ var Tabs = require('component/tabs/tabs');
+ var store = require('js/store');
```
There are some unused imports here (_, store)

bdeining replied:
Removed

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/tabs/search-form/tabs-search-form.view.js:
```diff
@@ -0,0 +1,39 @@
+ var $ = require('jquery');
+ var TabsView = require('../tabs.view');
+ var SearchFormModel = require('./tabs-search-form');
+ var store = require('js/store');
```
There are some unused imports (_, store, $)

bdeining replied:
Store is used on line 24

bdeining replied:
Removing the others

bdeining replied:
Removed

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/tabs/search-form/tabs-search-form.view.js:
```diff
@@ -0,0 +1,39 @@
+     },
+     initialize: function(options){
+         this.setDefaultModel();
+         this.options = options;
```
We don't need to store this off, it can be access throughout the view through `this.options`

bdeining replied:
Removed

Schachte commented on catalog/ui/catalog-ui-search/src/main/webapp/component/filter-builder-search-form/filter-builder.view.js:
```diff
@@ -0,0 +1,224 @@
+                if (this.filterContents){
+                    this.filterContents.currentView.collection.reset();
+                    filters.forEach(function (filter) {
+                        console.log(filter);
```
Remove

bdeining replied:
removed

Schachte commented on catalog/ui/catalog-ui-search/src/main/webapp/component/query-custom/query-custom.view.js:
```diff
@@ -0,0 +1,69 @@
+            store, QuerySettingsView, QueryAdvanced) {
+
+    return QueryAdvanced.extend({
+
```
Add `template: template,` here or else the `custom-view` handlebars template will never be used. Currently will inherit the template from the extended class.

bdeining replied:
added

bdeining commented:
Latest Screens : 
![screen shot 2018-03-14 at 3 49 44 pm](https://user-images.githubusercontent.com/6551038/37433082-2e56d81e-27a0-11e8-9694-7bcfc5b66b21.png)
![screen shot 2018-03-14 at 3 49 54 pm](https://user-images.githubusercontent.com/6551038/37433083-2e73b164-27a0-11e8-8cdc-72d3fd305174.png)
![screen shot 2018-03-14 at 3 50 06 pm](https://user-images.githubusercontent.com/6551038/37433084-2e91a1b0-27a0-11e8-8fb4-51b502c9cb0e.png)

mackncheesiest commented on ui/packages/catalog-ui-search/src/main/webapp/component/filter-builder-search-form/filter-builder.view.js:
```diff
@@ -0,0 +1,70 @@
+ *
+ **/
+/*global define, alert, setTimeout*/
+define([
```
✏️ Could refactor this to use the more maintainable `require` notation

bdeining replied:
done

mackncheesiest commented on ui/packages/catalog-ui-search/src/main/webapp/component/filter-search-form/filter.view.js:
```diff
@@ -0,0 +1,310 @@
+ *
+ **/
+/*global define, alert, setTimeout*/
+define([
```
✏️ Same nit about `requires`

bdeining replied:
done

bdeining commented on ui/packages/catalog-ui-search/src/main/webapp/component/search-form-selector/search-form-selector.view.js:
```diff
@@ -0,0 +1,35 @@
+var SearchFormsView = require('component/tabs/search-form/tabs-search-form.view');
+
+module.exports = Marionette.LayoutView.extend({
+    tagName: CustomElements.register('search-form-system'),
```
`search-form-selector`

bdeining replied:
done

bdeining commented on ui/packages/catalog-ui-search/src/main/webapp/component/search-form/search-form.less:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    .search-form-label {
+        font-weight: bold;
```
`.is-bold`

bdeining replied:
done

bdeining commented on ui/packages/catalog-ui-search/src/main/webapp/component/search-form/search-form.less:
```diff
@@ -0,0 +1,33 @@
+}
+
+@{customElementNamespace}search-form-loading {
+    padding: 2*@minimumSpacing;
```
whitespace

bdeining replied:
done

bdeining commented on ui/packages/catalog-ui-search/src/main/webapp/component/search-interactions/search-interactions.view.js:
```diff
@@ -32,34 +32,19 @@ module.exports = Marionette.LayoutView.extend({
+
     },
     onRender: function(){
+        this.listenTo(this.model, 'closeDropDown', this.triggerCloseDropdown);
```
listen to query model here instead

bdeining replied:
done

mackncheesiest commented on ui/packages/catalog-ui-search/src/main/webapp/component/query-custom/query-custom.view.js:
```diff
@@ -0,0 +1,69 @@
+    './query-custom.hbs',
+    'js/CustomElements',
+    'component/filter-builder-search-form/filter-builder.view',
+    'component/filter-builder/filter-builder',
```
❓ Wait why is the view coming from `filter-builder-search-form` and the model from `filter-builder`? Where does `component/filter-builder/filter-builder/view.js` factor into this?

bdeining replied:
The model need not be extended; the views use the same model but are presented a little differently.  `component/filter-builder/filter-builder.view.js` is used in the `query-advanced` view

bdeining commented:
build now

bdeining commented:
build now

jlcsmith commented:
Looks good.  There is a minor issue in IE11/Edge with a scroll bar on the text
![text-template-ie11edge](https://user-images.githubusercontent.com/4503545/38141013-6a101e40-3404-11e8-8cd4-ae5286869c31.PNG)
